flytekit/extras/external_container_sqlite3/task.py

```diff
@@ -0,0 +1,115 @@
+import contextlib
+import os
+import shutil
+import sqlite3
+import tempfile
+import typing
+from dataclasses import dataclass
+
+import pandas as pd
+
+from flytekit import FlyteContext, kwtypes
+from flytekit.core.base_sql_task import SQLTask
+from flytekit.core.context_manager import SerializationSettings
+from flytekit.core.python_third_party_task import PythonThirdPartyContainerTask
+from flytekit.core.shim_task import ShimTaskExecutor
+from flytekit.models import task as task_models
+from flytekit.types.schema import FlyteSchema
+
+
+def unarchive_file(local_path: str, to_dir: str):
+    """
+    Unarchive given archive and returns the unarchived file name. It is expected that only one file is unarchived.
+    More than one file or 0 files will result in a ``RuntimeError``
+    """
+    archive_dir = os.path.join(to_dir, "_arch")
+    shutil.unpack_archive(local_path, archive_dir)
+    # file gets uncompressed into to_dir/_arch/*.*
+    files = os.listdir(archive_dir)
+    if not files or len(files) == 0 or len(files) > 1:
+        raise RuntimeError(f"Uncompressed archive should contain only one file - found {files}!")
+    return os.path.join(archive_dir, files[0])
+
+
+@dataclass
+class SQLite3Config(object):
```

Comment:
How about use [asciiflow](https://asciiflow.com/#/) to explain whats happening in a SQLite3Task or just in general for a ThirdPartContainer task?